### ISSUE TITLE
review IDMapper

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#2468]
make advanced:
-ignore_charge

change default:
-mz_reference In my experience, mapping using the 'precursor' instead of the 'peptide' (theoretical mass) is inferior. I would vote for 'peptide' as  default,
- feature:use_centroid_mz should be default in my view as mapping to convex hulls leads to many false positives (also default for consensusXML?)

for reference:
```
Options (mandatory options marked with '*'):
  -id <file>*                  Protein/peptide identifications file (valid formats: 'mzid', 'idXML')
  -in <file>*                  Feature map/consensus map file (valid formats: 'featureXML', 'consensusXML', 'mzq')
  -out <file>*                 Output file (the format depends on the input file format). (valid formats: 'featureXML', 'consensusXML', 'mzq')

  -rt_tolerance <value>        RT tolerance (in seconds) for the matching of peptide identifications and (consensus) features. Tolerance is understood as 'plus or minus x', so the matching range increases by twice the given value. (default: '5' min: '0')
  -mz_tolerance <value>        M/z tolerance (in ppm or Da) for the matching of peptide identifications and (consensus) features. Tolerance is understood as 'plus or minus x', so the matching range increases by twice the given value. (default: '20' min: '0')
  -mz_measure <choice>         Unit of 'mz_tolerance'. (default: 'ppm' valid: 'ppm', 'Da')
  -mz_reference <choice>       Source of m/z values for peptide identifications. If 'precursor', the precursor-m/z from the idXML is used. If 'peptide', masses are computed from the sequences of peptide hits; in this case, an identification matches if any of its hits matches. ('peptide' should be used together with 'feature:use_centroid_mz' to avoid false-positive matches.) (default: 'precursor' valid: 'precursor', 'peptide')
  -ignore_charge               For feature/consensus maps: Assign an ID independently of whether its charge state matches that of the (consensus) feature.


Additional options for featureXML input:
  -feature:use_centroid_rt     Use the RT coordinates of the feature centroids for matching, instead of the RT ranges of the features/mass traces.
  -feature:use_centroid_mz     Use the m/z coordinates of the feature centroids for matching, instead of the m/z ranges of the features/mass traces. (If you choose 'peptide' as 'mz_reference', you should usually set this flag to avoid false-positive matches.)
```

**Comment by timosachsenberg** [comment]
@hendrikweisser @cbielow @jpfeuffer what is your experience with the mz_reference parameter?

Especially the default -feature:use_centroid_mz = false seems wrong/dangerous to me. At least from the documentation, it sounds like matching is performed to the full feature area (aka hundreds of ppms). What would make sense IMHO is to either just match to the precursor (option true) or additionally (option false) to the position of the isotopic traces.

**Comment by hendrikweisser** [comment]
I agree that the defaults for "mz_reference" and "use_centroid_mz" should be changed. If I remember correctly, they were left like this for backwards compatibility when matching by peptide mass was added.

> At least from the documentation, it sounds like matching is performed to the full feature area (aka hundreds of ppms). What would make sense IMHO is to either just match to the precursor (option true) or additionally (option false) to the position of the isotopic traces.

No, I'm pretty sure it is matching against the isotopic mass traces already. At least if they are available; if they aren't, it might use the whole feature area, which obviously isn't ideal, but then again the option to use only the centroid m/z already exists.

We could sacrifice a bit of flexibility and combine the "mz_reference" and "use_centroid_mz" parameters, i.e. automatically set "use_centroid_mz" sensibly depending on the choice of reference. I can't think of a use case where the "wrong" parameter combination would make sense.

**Comment by hendrikweisser** [comment]
> I can't think of a use case where the "wrong" parameter combination would make sense.

Wait, actually, one thing to keep in mind: If you don't perform precursor mass correction of MS2 spectra, what happens quite commonly is that MS2s from the M+1 isotopic peak get (slightly) misidentified as deamidated peptides. Since the identified sequence can still be correct, just with a spurious deamidation, the ID would still be usable (e.g. for protein inference/quantification). However, the ID wouldn't match to the monoisotopic peak (feature centroid) when matching by peptide mass.

That said, it would make sense to add an (optional) special case for matching of e.g. deamidated peptides, rather than allowing suboptimal settings just to cover this case.

**Comment by timosachsenberg** [comment]
I agree that it would make sense to extend the mapper to compensate for isotopic misassignments (which should be rather easy to add).

**Comment by lars20070** [comment]
I agree with the three changes as suggested by @timosachsenberg .
```
ignore_charge = false (default, advanced)
mz_reference = peptide (default)
use_centroid_mz = true (default)
```

**Comment by lars20070** [comment]
In my experience, the follow case is quite common and dangerous: The monoisotopic peak of peptide B corresponds (by chance) to the first isotopic peak of peptide A. An isotope-error tolerant search results in `ELVIS` and is incorrectly mapped to B.

Using `HighResPrecursorMassCorrector` before the search prevents the incorrect mapping and results in more/faster IDs. In any case (using the precursor correction or not), `mz_reference = peptide` is the saver setting.

![paper proteomics sketch book 11](https://user-images.githubusercontent.com/6149805/48063022-0ba4f600-e1c4-11e8-85ae-dcb2ca911442.jpeg)
